src/lib/utils/package-manifest.ts

```diff
@@ -140,32 +139,35 @@ function getTsSourceFromJsSource(
         return undefined;
     }
     if (
-        !hasOwnProperty(sourceMap, "sourceRoot") ||
-        !(typeof sourceMap.sourceRoot === "string") ||
         !hasOwnProperty(sourceMap, "sources") ||
         !Array.isArray(sourceMap.sources)
     ) {
         logger.error(
-            `The source map ${resolvedSourceMapURL} does not contain both "sourceRoot" and "sources".`
+            `The source map ${resolvedSourceMapURL} does not contain "sources".`
         );
         return undefined;
     }
+    let sourceRoot: string | undefined;
+    if (
+        hasOwnProperty(sourceMap, "sourceRoot") &&
+        typeof sourceMap.sourceRoot === "string"
+    ) {
+        sourceRoot = sourceMap.sourceRoot;
+    }
     // There's a pretty large assumption in here that we only have
     // 1 source file per js file. This is a pretty standard typescript approach,
     // but people might do interesting things with transpilation that could break this.
-    const sourcePath = resolve(
-        resolvedSourceMapURL,
-        "..",
-        sourceMap.sourceRoot,
-        sourceMap.sources[0]
-    );
+    let source = sourceMap.sources[0];
+    // If we have a sourceRoot, trim any leading slash from the source, and join them
+    // Similar to how it's done at https://github.com/mozilla/source-map/blob/58819f09018d56ef84dc41ba9c93f554e0645169/lib/util.js#L412
+    if (sourceRoot !== undefined) {
```

Comment:
Source maps don't _have_ to contain a sourceRoot and we handle that now. I also got this a little more conformant with the spec.